src/plugins/data_source_management/public/components/data_source_aggregated_view/data_source_aggregated_view.tsx

```diff
@@ -143,7 +146,12 @@ export class DataSourceAggregatedView extends React.Component<
 
   render() {
     if (this.state.showEmptyState) {
-      return <NoDataSource application={this.props.application} />;
+      return (
+        <NoDataSource
+          application={this.props.application}
+          hasIncompatibleDatasources={this.state.hasIncompatibleDataSources}
```

Comment:
What if we use `this.props.hasIncompatibleDataSources` instead of getting it from `state`

Reply:
The reason it uses a `state` prop is that the component can only determine if there are incompatible data sources _after_ it has already fetched the datasources.